tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -124,7 +151,7 @@
   private TonySession.Builder sessionBuilder;
 
   /** Configuration **/
-  private Configuration yarnConf;
+  private static Configuration yarnConf;
```

Comment:
Why did you make `yarnConf` `static`?

Reply:
Oh it was because calling it inside `createMetadataObj` and that was being called in `main` and `main` is static.

Reply:
I think this could be reverted back to non-static once I moved all the methods that handles history files to `Utils`.

Reply:
I think createHistoryFile() should be a non-static method in TonyApplicationMaster, meaning it should be called as `am.createHistoryFile()`.

`yarnConf` shouldn't be static because it should be associated with an instance of TonyApplicationMaster rather than be a single global class variable.

Reply:
Yeah, I agreed. I think it should be the same for `createMetadataObj` and `generateFileName` as these methods are specifically for creating the history file. Wdyt @erwa @oliverhu?

Reply:
After discussing offline with @oliverhu, we've decided to move these methods to a separate class `HistoryUtils` because TonyApplicationMaster is getting quite large, and it would be hard to maintain in the future. 
Eventually, we should move all utils (`Utils`, `ProtoUtils`, `HistoryUtils`, etc.) to a `Utils` package, but that could pushed to future PRs.

Reply:
Fixed: https://github.com/linkedin/TonY/pull/79/commits/16dbc9b453a5eeb6601bffd6bb5877909a329114.